### PR TITLE
makes the code more tdd friendly

### PR DESCRIPTION
**Pull request by ericminio**
To protect the production database and ease tdd, we allow a db to be injected.
A test db is ready and used in one test.
other side improvements: prettier, ci, badge

### CODE REVIEW

**Review by rxtphan — APPROVED** [review]
This is a great change @ericminio!  No feedback on the PR itself.  My only ask is that we inform the interviewing group of the changes so they can review them and understand the changes.  Also I don't think we need to adjust the candidate [instructions](https://docs.google.com/document/d/1KqROUOCBBgNhQ5HWQWHYX67V_9099lYiZXv4zI0pSFI/edit#heading=h.7v4v2cofhheu) but please let me know if you think we do.

File: routes/__tests__/petitions.test.js [hunk 1]
```diff
@@ -1,9 +1,49 @@
...
+        app.db = await getConnection(
+            `${__dirname}/../../db/node_api_exercise_tests.sqlite`
+        );
+    });
+    test('it is available', async () => {
+        const response = await request(app).get('/petitions');
+        expect(response.status).toBe(200);
+    });
+
+    test('it returns all the petitions', async () => {
```
Comment (ginayao1995) [inline routes/__tests__/petitions.test.js:16]: In the Rails version there’s no `returns all petitions test` written for them, and I like that it lets the candidate decide how they want to test (interact with the provided test database, vs importing a library).

What are your thoughts on leaving in the import and `beforeEach`, but removing it returns all the petitions? Having the import should be a hint to candidates that there is a test DB for them to interact with.

File: routes/__tests__/petitions.test.js [hunk 2]
```diff
@@ -1,9 +1,49 @@
...
+        await runQuery(app.db, 'delete from petitions;');
+        await runQuery(
+            app.db,
+            `
+      insert into petitions (starter_urn, title, body, created_at, updated_at)
+      values ('urn:changeorg:starter:1','Save our oceans','We need that','2020-01-01 00:00:00','2024-01-01 00:00:00')`
```
Comment (ginayao1995) [inline routes/__tests__/petitions.test.js:22]: Also in Rails are able to just do `Petition.create(...)` for example, as a next iteration after this is merged, we should talk about using an ORM & create empty models. Then in the tests the users wouldn't need to write SQL.

https://bookshelfjs.org/ is an option?

File: routes/__tests__/petitions.test.js (hunk 1 above)
Comment (ericminio) [reply routes/__tests__/petitions.test.js:16]: Removing the test also works for me.
Note that the code needs the db injection that happens in the `beforeEach`. 
Could be moved (hidden) into jest async config.

Comment (ericminio) [reply routes/__tests__/petitions.test.js:11]: Did that ^

File: README.md [hunk 3]
```diff
@@ -17,9 +19,10 @@ $ npm start
 
 The app will then be available at http://127.0.0.1:3000/
 
-There are also unit tests that you can run with
+## Working
```
Comment (ginayao1995) [inline README.md:22]: nit: "Working with tests"?

**Review by ginayao1995 — APPROVED** [review]

**Pull request merged**